lib/workload/stateless/stacks/icav2-copy-batch-utility/lambdas/check_or_launch_job/handler.py

```diff
@@ -0,0 +1,324 @@
+#!/usr/bin/env python3
+
+"""
+Lambda to determine if a given ICAv2 Copy Job has finished.
+Returns the status of the job which is one of the following
+* INITIALIZED
+* WAITING_FOR_RESOURCES
+* RUNNING
+* STOPPED
+* SUCCEEDED
+* PARTIALLY_SUCCEEDED
+* FAILED
+
+The event input is
+{
+    "dest_uri": "icav2://path/to/destination/folder/"
+    "source_uris": [
+        "icav2://path/to/data",
+        "icav2://path/to/data2",
+    ]
+    "job_id": null  # Or the job id abcd-1234-efgh-5678
+    "failed_job_list": []  # Empty list or list of failed jobs
+    "job_status": One of RUNNING, SUCCEEDED or FAILED (not the same as the job states, we reu)
+    "wait_time_seconds": int  # Number of seconds to wait before checking the job status - we double this each time we go through this loop
+}
+
+"""
+from pathlib import Path
+from typing import List
+
+# Wrapica imports
+from wrapica.job import get_job
+
+# Local imports
+from icav2_copy_batch_utility_tools.utils.job_helpers import submit_copy_job
+from icav2_copy_batch_utility_tools.utils.aws_ssm_helpers import set_icav2_env_vars
+
+# Globals
+SUCCESS_STATES = [
+    "SUCCEEDED"
+]
+TERMINAL_STATES = [
+    "STOPPED",
+    "FAILED",
+    "PARTIALLY_SUCCEEDED"
+]
+RUNNING_STATES = [
+    "INITIALIZED",
+    "WAITING_FOR_RESOURCES",
+    "RUNNING"
+]
+
+
+# Try a job 10 times before giving up
+MAX_JOB_ATTEMPT_COUNTER = 10
+DEFAULT_WAIT_TIME_SECONDS = 10
+
+
+def handler(event, context):
```

Comment:
Non blocker comment (due to time constraint). 

One fine day down the track, we should unit test this handler to guardrail couple of `if` branching statements down under. Need to make sure these branching logics maintain throughout the system SDLC and, retain it when refactoring happens.